search/utils.py

```diff
@@ -282,12 +331,18 @@ def create_mapping():
     """
 
     # Create the index if it doesn't exist.
-    conn = get_conn()
-    if not conn.indices.exists(INDEX_NAME):
-        conn.indices.create(INDEX_NAME)
+    get_conn()
+
+
+def _create_mapping():
```

Comment:
`_create_mapping` can only ever be called when `_CONN` is populated.

This is a workaround caused by `get_conn` in the old `create_mapping` being a source of recursion (`create_mapping` called `get_conn`, which called `create_mapping`, etc.) This was to fix the issue you reported where, under unusual circumstances, the mapping didn't exist.

The more straightforward solution would have been to rename `create_mapping` to `_create_mapping` and make it an internal call, only ever called by `_get_conn`. However, I couldn't do that because the `0001` migration in the search app calls `create_mapping` directly and it has been deployed already, so it probably shouldn't be changed. If migration `0001` had not yet been deployed, I would have changed its call to `create_mapping` to simply `get_conn`, which would have handled everything.